File: Python/ByTier/B/2162_B_Beautiful_String.py
# Problem: CF 2162 B - Beautiful String
# https://codeforces.com/contest/2162/problem/B

import sys

""" 
Algorithm: 
- The key insight is to construct a non-decreasing subsequence p such that when p is removed from s, the remaining string x is a palindrome.
- A simple and effective approach: take all '0's in the string to form the subsequence p. 
- Since '0' <= '1', this subsequence p is automatically non-decreasing.
- When we remove all '0's from s, we are left with all '1's, which form a string that is trivially a palindrome (since all characters are identical).
- If there are no '0's in the string, then removing an empty subsequence leaves the entire string, which is also a palindrome.

Time Complexity: O(n) where n is the length of the string.
Space Complexity: O(n) for storing the result indices.
"""

input = sys.stdin.readline

for _ in range(int(input())):
    n = int(input())
    s = input().strip()
    result = []
    for i in range(n):
        if s[i] == "0":
            result.append(i + 1)  # Store 1-based indices
    print(len(result))
    print(*result)


# https://github.com/VaHiX/CodeForces/

File: CPP/ByRound/2162/2162_B_Beautiful_String.cpp
// Problem: CF 2162 B - Beautiful String
// https://codeforces.com/contest/2162/problem/B

/*
Algorithm: Greedy approach
- The key insight is that for a subsequence to be non-decreasing in a binary string,
  it must consist of all '0's followed by all '1's (or just '0's, or just '1's).
- We want to remove a subsequence such that the remaining string is a palindrome.
- A good strategy is to take all the '1's in order to form a non-decreasing subsequence.
  When we remove all '1's, the remaining string consists of only '0's, which is always a palindrome.
- This greedy choice works because:
  1. The subsequence of '1's is non-decreasing.
  2. After removing the '1's, what remains is a string of '0's, which is trivially a palindrome.

Time Complexity: O(n) per test case
Space Complexity: O(n) for storing indices

The code:
- Reads input string s
- For each '1' in the string, stores its 1-based index in array b
- Outputs the count of '1's and their indices
*/
#include <iostream>
#include <string>

using namespace std;
constexpr int MAXN = 1e3 + 1;
int b[MAXN];
int main() {
  ios::sync_with_stdio(0);
  cin.tie(0);
  int t;
  cin >> t;
  while (t--) {
    int n;
    string s;
    cin >> n >> s;
    int k = 0;
    for (int i = 0; i < n; i++)
      if (s[i] == '1')           // If character is '1', store its index
        b[k++] = i + 1;          // Store 1-based index
    cout << k << '\n';           // Output number of '1's
    for (int i = 0; i < k; i++) {
      cout << b[i];
      if (i < k - 1)
        cout << ' ';             // Print space between indices
    }
    cout << '\n';
  }
}


// https://github.com/VaHiX/CodeForces/